### PR TITLE
[git-webkit] Derive branch name from bug title

### PR DESCRIPTION
**Pull request by JonWBedard**
#### 05aa010fdeed2809d25bb62c33d393aa9899409d
<pre>
[git-webkit] Derive branch name from bug title
<a href="https://bugs.webkit.org/show_bug.cgi?id=237284">https://bugs.webkit.org/show_bug.cgi?id=237284</a>
&lt;rdar://problem/89578241  &gt;

Reviewed by Dewei Zhu.

* Tools/Scripts/libraries/webkitscmpy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py:
(Branch.to_branch_name): Convert string (particularly a bug title) to potential branch name.
(Branch.main): If provided branch name is actually a bug number or URL, use the title
of that bug as the branch name.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/branch_unittest.py:
(TestBranch):
(TestBranch.test_prompt_git):
(TestBranch.test_prompt_number):
(TestBranch.test_prompt_url):
(TestBranch.test_invalid_branch):
(TestBranch.test_to_branch_name):

Canonical link: <a href="https://commits.webkit.org/247992@main">https://commits.webkit.org/247992@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@290748">https://svn.webkit.org/repository/webkit/trunk@290748</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py [hunk 1]
```diff
@@ -69,14 +70,34 @@ def branch_point(cls, repository):
 
         return commit
 
+    @classmethod
+    def to_branch_name(cls, value):
+        result = ''
+        for c in string_utils.decode(value):
+            if c in [u'-', u' ', u'\n', u'\t']:
+                result += '-'
```
Comment (dewei-zhu) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py]: Does this mean we may have multiple '-' if we have multiple spaces?

**Review by dewei-zhu — APPROVED** [review]
r=me

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py (hunk 1 above)
Comment (JonWBedard) [reply Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py]: Yes. Which seems ok, it feels like multiple consecutive spaces in a bug title is probably a mistake to being with.

**Comment by JonWBedard** [comment]
Landed https://commits.webkit.org/247992@main (05aa010fdeed)!

**Pull request merged**